packages/app-check/src/api.ts

```diff
@@ -52,10 +60,29 @@ export function activate(
   }
 
   const newState: AppCheckState = { ...state, activated: true };
+
+  // Read cached token from storage if it exists and store it in memory.
+  newState.cachedTokenPromise = readTokenFromStorage(app).then(cachedToken => {
+    if (cachedToken && isValid(cachedToken)) {
+      newState.token = cachedToken;
+    }
+    return cachedToken;
```

Comment:
You should use `getState()` and `setState()` since it's async, the `newState` here could already be outdated.

Reply:
Done.